### ISSUE TITLE
Analytics Implementation or QA Support for [VA Notify]

### ISSUE THREAD

**Issue by christy-tongty** [department-of-veterans-affairs/va.gov-team#65762]
# Google Analytics Implementation or QA Request

## What this form is for

<details>
<summary>More Information</summary>
<div>
    <br>
<p>Use this template to request Google Analytics implementation or QA for your product. Please fill out all brackets.</p>
    
> _Please see the following information for implementation timeline:_
> New Teams: Implementation will usually be completed in 4 sprints
> 
> - Sprint 1: Discovery/Orientation
> - Sprint 2: FE Implementation
> - Sprint 3: Tagging Implementation
> - Sprint 4: QA
> 
> All other teams: Implementation will usually be completed in 2 sprints
> 
> - Sprint 1: Discovery and FE Implementation
> - Sprint 2: Tagging Implementation and QA

</div>
</details>

### Required
- **Is this request for **Google Analytics implementation** or **QA**?**

     **- [x] Google Analytics implementation**
     - [ ] QA

- **Who is this request for?**

    - Team: VA Notify
    - Points of contact: 
        **- @christy-tongty  Christy Tong (VA Digital Corps Product Fellow) - christy.tong@va.gov
        - @GitSamJennings Samantha Jennings  (PM) - samantha.jennings@oddball.io]
        - @bevnobev Beverly Nelson (VA PO) - beverly.nelson@va.gov**
    
- **Provide the link(s) for Google Analytics implementation or QA**
  
    **- We've an existing[ GA Page](https://analytics.google.com/analytics/web/#/p310725521/reports/intelligenthome), that we created through this past [ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/39452). We're tracking our two environments (Production and Staging) and now would like to break up the [GA Page] (https://analytics.google.com/analytics/web/#/p310725521/reports/intelligenthome) into two separate GA pages - one for tracking each environment respectively: 
        - Production: https://notifications.va.gov -->'VA Notify - Production'
        - Staging: https://staging.notifications.va.gov -->'VA Notify - Staging'**
    
- **When is the launch date?**
    **- Both environments are live.** 

- **What interactions need to be tracked?** _NOTE: [Design System Components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Design-System-Components.1840808040.html) are automatically tracked_
    
    **- To include what we've been tracking on existing[ GA Page ](https://analytics.google.com/analytics/web/#/p310725521/reports/intelligenthome), might add further metrics later. We would like to separate the tracking of two environments (Production and Staging) as the first step.**
    
    - [If this is a funnel, insert each step's URL - pages visible depending on the applicant's answers should be marked accordingly]
    
    - [Insert other interactions - please review the [Design System Components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Design-System-Components.1840808040.html) and [GA Data Dictionary](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Google-Analytics-Data-Dictionary.1810464912.html)  to make sure these interactions aren't already tracked]

### Optional
- **If applicable, what test users are needed?**  _NOTE: Do not put staging credentials in this ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository_

    - [Insert test user]

- **Any additional comments**

    - [Insert comments]

> Please leave the following blank

## Acceptance criteria
- [ ] All appropriate issue tagging is completed
- [ ] QA has been completed
- [ ] Events have been added to `Product` and `Benefit Hub` Content Groups

## Additional information

Please refer to [Platform Collaboration Cycle](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/index.html) or the [Analytics Request Touchpoint](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Analytics-request.1782120453.html) on Platform website for more information about the Collaboration Cycle.

**Comment by christy-tongty** [comment]
@bevnobev @GitSamJennings Fyr - this is the Analytics implementation ticket to split our [GA Page] (https://analytics.google.com/analytics/web/#/p310725521/reports/intelligenthome) into tracking two environments separately.

**Comment by michelle-dooley** [comment]
@christy-tongty - yes we can break that into 2 views staging and prod. However we are still working on the VA.gov GA4 migration and would caution you in using that data until the work is complete.  As mentioned in original tickets b/c this is an internal site your team will be responsible for the tag/tracking management, has someone been doing that as part of this original set up? 
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdWUxIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--569fe4e2212ae6c1dc8926531ace7bc1ba406bdb/image.png)

**Comment by christy-tongty** [comment]
@michelle-dooley , we haven't done much in regards of GTM. Is there any set up we should put in place before splitting up the GA page, or can we begin the tagging after? Our Staging and Prod interfaces are not exactly in parity, and have some different events we wanted to track to see how users (mostly internal) are using our platform. Thank you!

**Comment by christy-tongty** [comment]
@michelle-dooley @ChristineYoungZA and Jamie (sorry, I don't know Jamie's GH handle), per our chat at office hour, it's brought up that it'd be helpful to browse our SSUI platform (Production: https://notifications.va.gov | Staging: https://staging.notifications.va.gov) to get a sense of how the platform is looking like - Now that you should be able to log in to browse both environments with your PIV. 

When logged in, you may hoover to the top right corner and pick "Va.gov" from the "choose service" dropdown. That's where a user can create email email/SMS templates, while template creation is currently locked for further access. 

Let me know if you need anything else from us for this ticket. Thank you!

**Comment by christy-tongty** [comment]
Hello @michelle-dooley, just wanted to follow up to see if you could share an ETA for dividing up the GA views. Thank you!

**Comment by michelle-dooley** [comment]
Hey @christy-tongty we will bring this into our sprint starting tomorrow.

**Comment by jestutt** [comment]
@christy-tongty  We have successfully moved the staging.notifications.va.gov to the **[GA4] VANotify - Staging** property. 

Some items of note: 

- When I removed the staging data stream from the production property -- **[GA4] VANotify** -- it removed all of the staging data from the property too. Unexpected but now you know that only notification.va.gov dats is in **[GA4] VANotify**.

- It looks like something happened to the tracking on notification.va.gov around Sep 8-9 as there has been no data collected from the production site since then. You'll want to make sure that the tracking code is still on the site:

```
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-QG2GS7QN4Q"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-QG2GS7QN4Q');
</script>
```

- Since the VANotify site environments are using a Google Tag implementation for GA4 instead of GTM, you'll be working with your front-end team to implement any events directly into the code. They'll probably find [Google's Developer's Guide](https://developers.google.com/analytics/devguides/collection/ga4) useful.
